crates/interpreter/src/exec.rs

```diff
@@ -989,7 +989,7 @@ impl<'m> Thread<'m> {
     }
 
     fn values(&mut self) -> &mut Vec<Val> {
-        &mut self.label().values
+        &mut self.frame().labels_values
```

Comment:
We're changing a bit the semantics here. Ideally the return value is never swapped and only pushed/popped.